conn/osxkeychain.c

```diff
@@ -0,0 +1,209 @@
+/**
+ * @file
+ * Credential management via macOS Keychain
+ *
+ * @authors
+ * Copyright (C) 2022 Ramkumar Ramachandra <r@artagnon.com>
+ *
+ * @copyright
+ * This program is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License as published by the Free Software
+ * Foundation, either version 2 of the License, or (at your option) any later
+ * version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/**
+ * @page conn_osxkeychain Read/write macOS Keychain credentials
+ *
+ * Called by mutt_account_getuser() and mutt_user_getpass() as an alternative to
+ * storing/retrieving credentials in the configutation file, or entering them in
+ * the prompt repeatedly. A dialog will automatically pop up requesting permissions
+ * on macOS the very first time.
+ *
+ * Enabled using the configure option '--osxkeychain'.
+ */
+
+#include <Security/Security.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include "mutt/lib.h"
+#include "connaccount.h"
+#include "mutt_account.h"
+
+/**
+ * struct Credential - A bunch of pointers that act as a lens into ConnAccount,
+ *                     filtering out only the information relevant to keychain.
+ */
+struct Credential
+{
+  SecProtocolType protocol; ///< Connection type, e.g. HTTP
+  UInt16 port;              ///< Port to connect to
+  const char *host;         ///< Server to login to
+  const char *user;         ///< Username
+  const char *pass;         ///< Password
+};
+
+static struct Credential cred;
```

Comment:
What's the lifespan of `cred`?
What if the user has multiple `Account`s?

I see that is doesn't **own** the objects, `host`, etc...
What happens if the `Account` goes away?

Reply:
Let's not do this and have conn_account_to_cred return a Credential object instead.